### PR TITLE
petsc: update to 3.21.2; slepc: update to 3.21.1, fix the port

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

`slepc` does not build since it is too old. Update both ports.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
CI on macOS-13 is MPICH-related, we cannot do anything about that here.

**Comment by ryandesign** [comment]
> CI on macOS-13 is MPICH-related, we cannot do anything about that here.

Really? It says:

```
*********************************************************************************************
           UNABLE to CONFIGURE with GIVEN OPTIONS (see configure.log for details):
---------------------------------------------------------------------------------------------
  C compiler you provided with -CC=/opt/local/bin/mpicc-mpich-mp cannot be found or does
  not work.
  If the above linker messages do not indicate failure of the compiler you can rerun with
  the option --ignoreLinkOutput=1
*********************************************************************************************
```

We can't fix that?

**Comment by barracuda156** [comment]
@ryandesign Admittedly I did not download logs, but given it builds on macOS-14 and macOS-12, but specifically fails on macOS-13, this looks exactly like a breaking flag in MPICH PG, which breaks CI for every port using it.

I recall I have submitted a patch which fixed the problem, but there was no enthusiasm to merge it, so perhaps that PR was closed.

File: math/slepc/Portfile [hunk 1]
```diff
@@ -22,27 +22,36 @@ notes                       "Add the following lines to your .bash_profile if yo
...
+set python.version          [string map {. ""} $python_ver_dot]
 
 depends_build-append        port:sowing \
                             port:python${python.version}
 
+configure.python            ${prefix}/bin/python${python_ver_dot}
+
+post-patch {
+    reinplace "s|/usr/bin/env python3|${configure.python}|" ${worksrcpath}/configure
+    fs-traverse f ${worksrcpath} {
```
Comment (ryandesign) [inline math/slepc/Portfile:48]: The second argument to `fs-traverse` must be a list of paths:

```suggestion
    fs-traverse f [list ${worksrcpath}] {
```

**Comment by barracuda156** [comment]
@ryandesign Squashed and rebased now with your change.

**Pull request merged**